platforms/android/library/src/main/java/io/element/android/wysiwyg/EditorEditText.kt

```diff
@@ -232,6 +245,21 @@ class EditorEditText : AppCompatEditText {
         }
     }
 
+    /**
+     * Sets up the [HtmlConverter] used to translate HTML to Spanned text.
+     * @param styleConfig The styles to use for the generated spans.
+     * @param mentionDisplayHandler Used to decide how to display any mentions found in the HTML text.
+     */
+    fun setupHtmlConverter(styleConfig: StyleConfig, mentionDisplayHandler: MentionDisplayHandler?) {
```

Comment:
Users of the View library may have set the style config through XML so it should be possible to set the mention display handler while respecting this. Can we make the `styleConfig` optional so that this use case is not broken?

Reply:
If you use the current implementation, it *should* just re-set the same `styleConfig` it previously had unless I messed it up. This was the only way I could think of for reducing the number of times we re-create the `HtmlConverter` in each `update` recomposition.

Reply:
I'm not sure I understand - I was talking about apps that aren't using compose e.g. Element Android. The [removal](https://github.com/matrix-org/matrix-rich-text-editor/pull/854/files/18ed04108c0f31da539e07a686f0bf5de67eaf72#diff-0759d3c3c35a9122b7b81526317c7a65ba00cbc0dca69c3c0302c4e8c27d649dL99-L106) of the `var mentionDisplayHandler` means that they'll be unable to set this variable independently of the style (for which they may use XML).

> reducing the number of times we re-create the HtmlConverter in each `update` recomposition

Given that `update` only fires when one of the config variables (`style`, `typeface`, `onError` etc changes) it's unlikely to be called really frequently because of their nature. So IMO we don't need to over optimise it.

Reply:
> I'm not sure I understand - I was talking about apps that aren't using compose e.g. Element Android. The [removal](https://github.com/matrix-org/matrix-rich-text-editor/pull/854/files/18ed04108c0f31da539e07a686f0bf5de67eaf72#diff-0759d3c3c35a9122b7b81526317c7a65ba00cbc0dca69c3c0302c4e8c27d649dL99-L106) of the var mentionDisplayHandler means that they'll be unable to set this variable independently of the style (for which they may use XML).

Sorry, I think I wasn't clear enough. The `styleConfig` property's getter is now public, so they can just do:

```kotlin
editText.updateStyle(styleConfig = this.styleConfig, mentionDisplayHandler = mentionDisplayHandler)
```

It does feel a bit weird, but the alternative is having custom setters for the 2 properties, making `StyleConfig` nullable or checking with `this::styleConfig.isInitialized` and then creating a new `HtmlConverter` with the `styleConfig` already set and whatever `mentionDisplayHandler` is passed or was already there, making the logic harder to follow IMO.

> Given that update only fires when one of the config variables (style, typeface, onError etc) changes it's unlikely to be called really frequently because of their nature. So IMO we don't need to over optimise it.

It's true it won't be called that often, but it'll still create 2 `HtmlConverter` instances for each update, always discarding the first one, which is pointless.

Reply:
Ok, thanks for explaining, let's keep it this way.